core/types/l2trace.go

```diff
@@ -1,165 +1,78 @@
-	Failed      bool
-	ReturnValue string
-	StructLogs  []StructLogRes
-}
-
-func (e *ExecutionResult) EncodeRLP(w io.Writer) error {
-	return rlp.Encode(w, rlpExecutionResult{
-		Gas:         e.Gas,
-		Failed:      e.Failed,
-		ReturnValue: e.ReturnValue,
-		StructLogs:  e.StructLogs,
-	})
-}
-
-func (e *ExecutionResult) DecodeRLP(s *rlp.Stream) error {
-	var dec rlpExecutionResult
-	err := s.Decode(&dec)
-	if err == nil {
-		e.Gas, e.Failed, e.ReturnValue, e.StructLogs = dec.Gas, dec.Failed, dec.ReturnValue, dec.StructLogs
-	}
-	return err
+	Gas         uint64 `json:"gas"`
+	Failed      bool   `json:"failed"`
+	ReturnValue string `json:"returnValue,omitempty"`
+	// If it is a contract call, the contract code is returned.
+	ByteCode string `json:"byteCode,omitempty"`
+	// The account's proof.
+	Proof      []string       `json:"proof,omitempty"`
+	StructLogs []StructLogRes `json:"structLogs"`
 }
 
 // StructLogRes stores a structured log emitted by the EVM while replaying a
 // transaction in debug mode
 type StructLogRes struct {
-	Pc      uint64             `json:"pc"`
-	Op      string             `json:"op"`
-	Gas     uint64             `json:"gas"`
-	GasCost uint64             `json:"gasCost"`
-	Depth   int                `json:"depth"`
-	Error   string             `json:"error,omitempty"`
-	Stack   *[]string          `json:"stack,omitempty"`
-	Memory  *[]string          `json:"memory,omitempty"`
-	Storage *map[string]string `json:"storage,omitempty"`
+	Pc        uint64             `json:"pc"`
+	Op        string             `json:"op"`
+	Gas       uint64             `json:"gas"`
+	GasCost   uint64             `json:"gasCost"`
+	Depth     int                `json:"depth"`
+	Error     string             `json:"error,omitempty"`
+	Stack     *[]string          `json:"stack,omitempty"`
+	Memory    *[]string          `json:"memory,omitempty"`
+	Storage   *map[string]string `json:"storage,omitempty"`
+	ExtraData *ExtraData         `json:"extraData,omitempty"`
 }
 
-type rlpStructLogRes struct {
-	Pc      uint64
-	Op      string
-	Gas     uint64
-	GasCost uint64
-	Depth   uint
-	Error   string
-	Stack   []string
-	Memory  []string
-	Storage []string
+type ExtraData struct {
+	// CREATE: sender address
+	From *common.Address `json:"from,omitempty"`
+	// CREATE: sender nonce
+	Nonce *uint64 `json:"nonce,omitempty"`
+	// CALL | CALLCODE | DELEGATECALL | STATICCALL: [tx.to address’s code_hash, stack.nth_last(1) address’s code_hash]
+	CodeHashList []common.Hash `json:"codeHashList,omitempty"`
+	// SSTORE | SLOAD: [storageProof]
+	// SELFDESTRUCT: [contract address’s accountProof, stack.nth_last(0) address’s accountProof]
+	// SELFBALANCE: [contract address’s accountProof]
+	// BALANCE | EXTCODEHASH: [stack.nth_last(0) address’s accountProof]
+	// CREATE | CREATE2: [created contract address’s accountProof]
+	// CALL | CALLCODE: [caller contract address’s accountProof, callee contract address’s accountProof, stack.nth_last(1) address’s accountProof]
```

Comment:
`callee contract address` is `stack.nth_last(1) address`

`stack.nth_last(1) address` is `callee contract address`